### PR TITLE
YTP 8

### PR DESCRIPTION
**Pull request by fm-fede-ravchina**

### CODE REVIEW

**Review by fm-andres-rangel — CHANGES_REQUESTED** [review]

File: include/ytp/sequence.h [hunk 1]
```diff
@@ -464,35 +425,36 @@ FMMODFUNC void ytp_sequence_cb_rm(ytp_sequence_t *seq);
...
- * @param[out] error
+ * @param[out] error out-parameter for error handling
  * @return ytp_iterator_t
  *
  * Sets data offset to offset. All control messages from the
  * beginning up to the offset are guarantied do have been processed
  * and the callbacks to be called.
  */
-FMMODFUNC ytp_iterator_t ytp_sequence_seek(ytp_sequence_t *seq, size_t off,
+FMMODFUNC ytp_iterator_t ytp_sequence_seek(ytp_sequence_t *seq, uint64_t offset,
```
[the remark this answers is not in the document]
Comment (maxim-trokhimtchouk) [reply include/ytp/sequence.h:437]: change to ytp_mmnode_offs

File: src/ytp/api.cpp [hunk 2]
```diff
@@ -227,8 +212,7 @@ size_t ytp_sequence_shared_tell(shared_sequence *sh_seq,
 }
 
 static struct ytp_sequence_api_v1 api_v1 {
-  ytp_sequence_shared_reserve, ytp_sequence_shared_commit,
-      ytp_sequence_shared_sub, ytp_sequence_shared_dir,
+  ytp_sequence_shared_reserve, ytp_sequence_shared_commit, NULL, NULL,
```
[the remark this answers is not in the document]
Comment (maxim-trokhimtchouk) [reply src/ytp/api.cpp:215]: make ytp_sequence_shared_sub, ytp_sequence_shared_dir, not implemented

**Review by maxim-trokhimtchouk — CHANGES_REQUESTED** [review]
test syncer built with new ytp connected to syncer built with old ytp and the reverse.
performance test similar to before (yamal-perf, syncer perf, number of messages read per sec)

File: include/ytp/cursor.h [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                                     fmc_error_t **error);
+
+/**
+ * @brief Checks if there are not more messages
+ *
+ * @param[in] cursor the ytp_cursor_t object
+ * @param[out] error out-parameter for error handling
+ * @return true if there are not more messages, false otherwise
+ */
+FMMODFUNC bool ytp_cursor_term(ytp_cursor_t *cursor, fmc_error_t **error);
```
Comment (maxim-trokhimtchouk) [inline include/ytp/cursor.h:151]: maybe removing this

File: include/ytp/cursor.h [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+/**
+ * @brief Reads one message and executes the callbacks that applies.
+ *
+ * @param[in] cursor the ytp_cursor_t object
+ * @param[out] error out-parameter for error handling
+ * @return true if a message was processed, false otherwise
+ */
+FMMODFUNC bool ytp_cursor_poll(ytp_cursor_t *cursor, fmc_error_t **error);
```
Comment (maxim-trokhimtchouk) [inline include/ytp/cursor.h:160]: add TODO for ytp_cursor_poll_until

File: src/ytp/glob.cpp [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+#include <ytp/glob.h>
+
+#include <map>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "cursor.h"
+
+struct ytp_glob {
```
Comment (maxim-trokhimtchouk) [inline src/ytp/glob.cpp:27]: add a TODO note to properly handle exceptions

**Review by maxim-trokhimtchouk — CHANGES_REQUESTED** [review]

**Review by maxim-trokhimtchouk — DISMISSED** [review]

**Review by fm-andres-rangel — DISMISSED** [review]

**Review by fm-andres-rangel — APPROVED** [review]

**Review by fm-andres-rangel — APPROVED** [review]

**Pull request merged**